ui/plugin-system/src/components/PluginKindSelect/PluginKindSelect.tsx

```diff
@@ -12,31 +12,67 @@
 
-export interface PluginKindSelectProps extends Omit<TextFieldProps, 'children'> {
-  pluginType: PluginType;
+export interface PluginKindSelectProps extends Omit<TextFieldProps, 'value' | 'onChange' | 'children'> {
+  pluginTypes: PluginType[];
+  value?: PluginEditorSelection;
+  onChange?: (s: PluginEditorSelection) => void;
 }
 
 /**
- * Displays a MUI Select input for selecting a plugin's kind from a list of all the available plugins of a specific
- * plugin type. (e.g. "Show a list of all the Panel plugins" or "Show a list of all the Variable plugins").
+ * Displays a MUI Select input for selecting a plugin's kind from a list of all the available plugins of some specific
+ * plugin types. (e.g. "Show a list of all the Panel plugins", or "Show a list of all the Variable plugins", or "Show
+ * a list of all the TimeSeriesQuery, TraceQuery, and LogQuery plugins").
+ * The value of the select is the kind of the plugin, but you can also listen to the `onPluginTypeChange` event to know
+ * when the user changes the plugin type (it fires at start for the default value.)
  */
 export const PluginKindSelect = forwardRef((props: PluginKindSelectProps, ref) => {
-  const { pluginType, value: propValue, ...others } = props;
-  const { data, isLoading } = useListPluginMetadata(pluginType);
+  const { pluginTypes, value: propValue, onChange: propOnChange, ...others } = props;
+  const { data, isLoading } = useListPluginMetadata(pluginTypes);
 
   // Pass an empty value while options are still loading so MUI doesn't complain about us using an "out of range" value
-  const value = propValue !== '' && isLoading ? '' : propValue;
+  const value = propValue && isLoading ? '' : JSON.stringify(propValue);
+
+  const onChange = (event: { target: { value: string } }) => {
+    if (propOnChange === undefined) {
+      return;
+    }
+    propOnChange(JSON.parse(event.target.value));
+  };
+
+  const renderValue = useCallback(
+    (selected: unknown) => {
+      const selectedValue = JSON.parse(selected as string);
+      if (!selectedValue.kind) {
+        return '';
+      }
+      return data?.find((v) => v.pluginType === selectedValue.type && v.kind === selectedValue.kind)?.display.name;
+    },
+    [data]
+  );
 
   // TODO: Does this need a loading indicator of some kind?
   return (
-    <TextField select inputRef={ref} {...others} value={value} data-testid="plugin-kind-select">
+    <TextField
+      select
+      inputRef={ref}
+      {...others}
+      value={value}
+      onChange={onChange}
+      SelectProps={{ renderValue }}
+      data-testid="plugin-kind-select"
+    >
       {isLoading && <MenuItem value="">Loading...</MenuItem>}
       {data?.map((metadata) => (
-        <MenuItem data-testid="option" key={metadata.kind} value={metadata.kind}>
+        <MenuItem
+          data-testid="option"
+          key={metadata.pluginType + metadata.kind}
+          value={JSON.stringify({ type: metadata.pluginType, kind: metadata.kind })}
```

Comment:
as far as I understood, the react selector is not supporting objects as input, that's why I'm using this JSON.parse/stringify solution https://stackoverflow.com/a/49442580/5690223

Reply:
Okay checking twice this morning with an existing sandbox it seems to not be true anymore. https://codesandbox.io/p/sandbox/react-select-playground-pxm1f?file=%2Fsrc%2FApp.js%3A39%2C13

Let me try again

Reply:
I think we are using the TextField from MUI, which internally uses the material UI select which supports objects as options/values, https://mui.com/material-ui/api/select/#props

Reply:
Made a [sandbox](https://codesandbox.io/p/sandbox/dreamy-shadow-3xxwgq?file=%2Fsrc%2FApp.tsx%3A16%2C2&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clutiejwp00063b6j5zprdv6a%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clutiejwp00023b6j051fo9wp%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clutiejwp00033b6jbotxy5l1%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clutiejwp00053b6jklu2v040%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clutiejwp00023b6j051fo9wp%2522%253A%257B%2522id%2522%253A%2522clutiejwp00023b6j051fo9wp%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clutifvp100483b6jgs4zs3rs%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A16%252C%2522startColumn%2522%253A2%252C%2522endLineNumber%2522%253A16%252C%2522endColumn%2522%253A2%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clutifvp100483b6jgs4zs3rs%2522%257D%252C%2522clutiejwp00053b6jklu2v040%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clutiejwp00043b6j72rhax0e%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clutiejwp00053b6jklu2v040%2522%252C%2522activeTabId%2522%253A%2522clutiejwp00043b6j72rhax0e%2522%257D%252C%2522clutiejwp00033b6jbotxy5l1%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clutiejwp00033b6jbotxy5l1%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) with a test. As expected TypeScript complains because the MUI typing for this case is not generic enough, but it seems to be supported. If you have trouble with the types I'll be also ok with the current serializing solution, just a nit.

Reply:
Oh thank you for the sandbox! I just noticed it... The thing is with our config, any doesn't pass well the linter. I made an update, WDYT?

Reply:
it seems that a lot of scape hatches are needed for this simple typing to work. As said before, it will be better then to use the initial serialization solution. Sorry for the fuzz, too many scape hatches will make the code error prone and hard to read. I tried also locally, and there is no simple solution that just avoiding TypeScript. IMHO not issues with TypeScript itself or your code but the way this component closes extension :..(

Reply:
Okay no problem I can get back the original code ;)

Reply:
Updated to use back the serialization solution

Reply:
I'll mark it as resolved